Spokestack/Tokenizer.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  Tokenizer.swift
+//  Spokestack
+//
+//  Created by Noel Weichbrodt on 2/14/20.
+//  Copyright © 2020 Pylon AI, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A simple protocol for Tokenizer services used by NLUService instances.
+@objc public protocol Tokenizer {
```

Comment:
Is the Wordpiece encoder meant to adopt this protocol? If so, I'd suggest a closer correspondence with the Android version (which I suppose I haven't released yet, so I was just expecting you to read my mind there...). It wraps tokenization and token encoding (into IDs for feeding the model) into a single step for efficiency, which has the additional effect of making it not a tokenizer but more of a token encoder.

In general, though, a tokenizer isn't a necessary piece of the top-level NLU infrastructure, as almost any third-party service we'd provide a plugin for would expect a string and do tokenization/token encoding internally. I'd hide this down alongside the TF NLU implementation since Spokestack isn't really a general-purpose NLP library.